### PR TITLE
Updated multipath config to use mapper class

### PR DESCRIPTION
**Pull request by PaulWay**
Preparing to deprecate the mapper function 'multipath' with a MultipathConf mapper class.

Also added and formatted documentation including usage examples.

Tests cover 100% of code.

### CODE REVIEW

File: insights/parsers/multipath_conf.py [hunk 1]
```diff
@@ -1,42 +1,23 @@
-from insights.contrib import pyparsing as p
-from .. import parser
+"""
+MultipathConf - file ``/etc/multipath.conf``
```
Comment (bfahr) [inline insights/parsers/multipath_conf.py]: You are missing the underline on the title.

File: docs/shared_mappers_catalog/multipath_conf.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.. automodule:: insights.mappers.multipath_conf
```
Comment (bfahr) [inline docs/shared_mappers_catalog/multipath_conf.rst]: This file needs to be moved to insights/docs/shared_parsers_catalog, and import needs to be insights.parsers.multipath_conf.

File: insights/parsers/multipath_conf.py (hunk 1 above)
Comment (PaulWay) [inline insights/parsers/multipath_conf.py]: Dangit!  Dunno how I removed that.  Fixed!

File: docs/shared_mappers_catalog/multipath_conf.rst (hunk 2 above)
Comment (PaulWay) [inline docs/shared_mappers_catalog/multipath_conf.rst]: Yep, missed that in the rebase.  Fixed.  Thanks @bfahr

**Comment by xiangce** [comment]
Since Paul has already fixed the commented points, I will go ahead and merge this PR to `master`

**Pull request merged**